test_sets/new_vehicle_data_hands_off_steering.txt

```diff
@@ -0,0 +1,33 @@
+./test_scripts/API/VehicleData/HandsOffSteering/008_SubscribeVD_disallowed_after_PTU.lua
+;./test_scripts/API/VehicleData/HandsOffSteering/009_SubscribeVD_not_success_codes_for_handsOffSteering_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
+./test_scripts/API/VehicleData/HandsOffSteering/010_SubscribeVD_HMI_responds_with_invalid_data.lua
+./test_scripts/API/VehicleData/HandsOffSteering/011_SubscribeVD_App_sends_invalid_request.lua
+./test_scripts/API/VehicleData/HandsOffSteering/012_SubscribeVD_for_two_Apps.lua
+./test_scripts/API/VehicleData/HandsOffSteering/013_UnsubscribeVD_Success.lua
+./test_scripts/API/VehicleData/HandsOffSteering/014_UnsubscribeVD_disallowed.lua
+./test_scripts/API/VehicleData/HandsOffSteering/015_UnsubscribeVD_disallowed_after_PTU.lua
+;./test_scripts/API/VehicleData/HandsOffSteering/016_UnsubscribeVD_not_success_codes_for_handsOffSteering_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
+./test_scripts/API/VehicleData/HandsOffSteering/017_UnsubscribeVD_HMI_responds_with_invalid_data.lua
+./test_scripts/API/VehicleData/HandsOffSteering/018_UnsubscribeVD_App_sends_invalid_request.lua
+./test_scripts/API/VehicleData/HandsOffSteering/019_UnsubscribeVD_for_two_Apps.lua
+./test_scripts/API/VehicleData/HandsOffSteering/020_OnVD_notification_Success.lua
+;./test_scripts/API/VehicleData/HandsOffSteering/021_OnVD_notification_disallowed.lua https://github.com/smartdevicelink/sdl_core/issues/3365
+;./test_scripts/API/VehicleData/HandsOffSteering/022_OnVD_disallowed_after_PTU.lua https://github.com/smartdevicelink/sdl_core/issues/3365
+./test_scripts/API/VehicleData/HandsOffSteering/023_OnVD_notification_with_invalid_data.lua
+./test_scripts/API/VehicleData/HandsOffSteering/024_OnVD_notification_App_is_not_subscribed.lua
+./test_scripts/API/VehicleData/HandsOffSteering/025_OnVD_notification_for_two_Apps.lua
+./test_scripts/API/VehicleData/HandsOffSteering/026_handsOffSteering_parameter_processing_in_case_app_version_is_greater_than_parameter_version.lua
+./test_scripts/API/VehicleData/HandsOffSteering/027_handsOffSteering_parameter_processing_in_case_app_version_is_less_than_parameter_version.lua
+./test_scripts/API/VehicleData/HandsOffSteering/028_SubscribeVD_during_resumption_unexpected_disconnect.lua
+./test_scripts/API/VehicleData/HandsOffSteering/029_SubscribeVD_during_resumption_IGN_OFF_ON.lua
+;./test_scripts/API/VehicleData/HandsOffSteering/030_SubscribeVD_during_resumption_unexpected_disconnect_for_two_Apps.lua https://github.com/smartdevicelink/sdl_core/issues/2283
+;./test_scripts/API/VehicleData/HandsOffSteering/031_SubscribeVD_during_resumption_IGN_OFF_ON_for_two_Apps.lua https://github.com/smartdevicelink/sdl_core/issues/2283
+./test_scripts/API/VehicleData/HandsOffSteering/032_SubscribeVD_IGNORED_app_already_subscribed.lua
+./test_scripts/API/VehicleData/HandsOffSteering/033_UnsubscribeVD_IGNORED_app_already_unsubscribed.lua
```

Comment:
These tests don't seem to account for the other available vehicle data test PRs (such as https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2417). These tests would be much more maintainable if they tested multiple vehicle data items, since there aren't too many cases which are exclusive to a specific vehicle data type. In addition, the basis for these vehicle data tests already exists in 
1. `API/VehicleData/GetVehicleData`
2. `API/VehicleData/SubscribeVehicleData`
3. `API/VehicleData/UnsubscribeVehicleData`
4. `API/VehicleData/OnVehicleData`.

I'm all for testing these RPCs more thoroughly, but we should expand off of the existing tests and we should make sure this remains scalable so that we can cover all vehicle data items equally.

Reply:
@jacobkeeler We agree with your point. Having 'general' common approach for all VD proposals is much better. 
However it would have been possible if scope of `7.0` release had been known beforehand. 
Unfortunately this scope hadn't been finalized when these proposals were in development.
That's why we chose independent approach which would allow us to simple exclude one proposal if it would had been removed from scope.

Taking into account tight schedule of the release we may propose to create a maintenance PR in the future and to rework all existing scripts related to VD.

Reply:
@dboltovskyi Regardless of whether combining these feature tests was feasible, existing test cases should still be taken into account. Notably the following cases should be expanded rather than duplicated as they are here:

1. The `0XX_*VD_Success_flow.lua` tests are covered by the `*VehicleData/001_Success_flow.lua` tests 
2. The `0XX_*VD_disallowed_after_PTU.lua` tests are covered by the `*VehicleData/0XX_RPC_parameter_DISALLOWED_by_policies_flow.lua` tests

Please remove these new tests and expand upon the existing tests instead.

In addition, I don't think that this vehicle data item should have its own `common` file and it should instead use `commonVehicleData.lua`.

I still think that basing these tests off of a common test set would have been easier and quicker from the start (even if a feature was cut) but since these have already been made separately, it should at least be verified that the test coverage for each of these vehicle data items is the same. Test names between features should be consistent as well, this is not the case for https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2417 at the moment.

Reply:
@jacobkeeler 
We would like to introduce completely redesigned solution in c36b408 
Vehicle Data scripts in folders `GetVehicleData`, `OnVehicleData`, `SubscribeVehicleData` and `UnsubscribeVehicleData` as well as `common`module reworked.

Summary of changes:

 - All of the following checks are run against each of existing VD parameter defined in APIs
    - Success flow
    - Disallowed in PT flow
    - Disallowed after PTU flow
    - Invalid response from HMI
    - Invalid request from App
    - Response from HMI with only mandatory sub-parameters defined
    - Response from HMI with missing mandatory sub-parameter
    - Versioning: App version is greater/less than VD parameter version
    - Subscription for 1 or 2 Apps
    - Ignorance of 2nd subscription/unsubscription request
    - Resumption of subscription for 1 or 2 Apps in case of unexpected disconnect/ignition cycle
    - Response from HMI with not successful result codes
 - Specific checks related to some proposal moved to separate folder (e.g. `AudioSource`)
 - Valid values for each VD parameter generated automatically based on restrictions defined in API
 - Invalid values (type) for each VD parameter calculated automatically as well

There is a small update in ATF which is required for this solution: https://github.com/smartdevicelink/sdl_atf/pull/204

This approach will significantly reduce amount of work required to verify new VD parameter.
If such parameter is introduced the only change that is needed is to add this parameter into `m.vd` table in `common` module.
All the checks will be applied automatically.
Only specific scenarios will need to be covered. Fortunately usually there are only a few of them.

In case if this solution is accepted all other 3 VD proposals in review would require update as well.
And after that would contain much less changes.